docs/devel/owners.md

```diff
@@ -0,0 +1,126 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/devel/owners.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Owners files
+
+_Note_: This is a design for a feature that is not yet implemented.
+
+## Overview
+
+We want to establish owners for different parts of the code in the Kubernetes codebase.  These owners
+will serve as the approvers for code to be submitted to these parts of the repository.  Notably, owners
+are not necessarily expected to do the first code review for all commits to these areas, but they are
+required to approve changes before they can be merged.
+
+## High Level flow
+
+### Step One: A PR is submitted
+
+After a PR is submitted, a user can request the approval from the required owners.  The user requests this
+by writing:
+
+```
+@k8s-bot approvers
+```
+
+into a comment.  When our automation sees this comment, it will reply with an annotation that indicates
+the owners required to approve.  The annotation is a comment that is applied to the PR.
+This comment will say:
+
+```
+Approval is required from <owner-a> OR <owner-b> AND <owner-c> OR <owner-d> ...
+```
+
+The set of required authors is drawn from the OWNERS files in the repository (see below).  For each file
+there should be multiple different OWNERS, these owners are listed in the `OR` clause(s). Because
+it is possible that a PR may cover different directories, with disjoint sets of OWNERS, a PR may require
+approval from more than one person, this is where the `AND` clauses come from.
+
+PR authors are not required to ask for approval, if for example, the reviewer is already an owner, it is
+expected that the reviewer will (eventually) grant approval without being explicitly asked to do so.
```

Comment:
This sentence is written in a way that makes the meaning unclear. I'd suggest "There are some circumstances under which the PR author is not required to ask for approval. For example, if the PR author is an owner in all modified directories, or if the reviewer is an owner."

But I think it is easier to just automatically add the "Approval is required from..." line always, instead of requiring people to figure out whether they need to add "@k8s-bot approvers" , etc.